### ISSUE TITLE
[a11y][Modal] Activator focus management should be managed by the modal

### ISSUE THREAD

**Issue by chloerice** [Shopify/polaris-react#1932]
## Issue summary

An important best practice for all overlay components is to return focus to the activating element on closing the overlay. But currently, there is no support for ref forwarding of Polaris components. This forces focus of modal activators to be done by wrapping components like `Button` or `TextField` in a container with a ref. Most modals in `web` do not currently manage focus at all and this is a big accessibility issue. We should bake in focus management into `Modal` just as we do with `Popover`.

**Comment by chloerice** [comment]
Closing in favor of https://github.com/Shopify/polaris-react/issues/795

**Comment by dpersing** [comment]
@LauraAubin, I'm adding you as an assignee and moving this issue to In Progress, since it looks like you're tackling it with PR #2140 as part of the work on issue #795. Let me know if that's incorrect and I'll move it back to To do!

**Comment by LauraAubin** [comment]
Closing this since the modal now supports an activator, and we'll eventually make it mandatory at a later date.
